File: 07_Tetris_Game_Java/Driver2.java
import java.util.Scanner;
import java.util.Random;

public class Driver2 {
    public static void main(String[] args) {
        int row, column;

        String input;

        Tetromino.Shapes shape_arr[]= {Tetromino.Shapes.I,
            Tetromino.Shapes.O,Tetromino.Shapes.T,
            Tetromino.Shapes.J,Tetromino.Shapes.L, 
            Tetromino.Shapes.S, Tetromino.Shapes.Z }; //random array

        
        System.out.print("Enter row of the board: ");
        Scanner sc = new Scanner(System.in);

        row = sc.nextInt();

        System.out.print("Enter column of the board: ");
        column = sc.nextInt();

        System.out.println("Game board is creating... ");
        Tetris tetBoard = new Tetris(row, column);

        boolean flag = false;
        while(!tetBoard.isGameOver() || flag){
            flag = false;

            System.out.print("Enter the tetromino type(I,O,T,J,L,S,Z or R for random, Q for Quit.): ");
            input = sc.next();
            System.out.println(input);

            Tetromino.Shapes myShape;

            if(input.equals("Q")) {
                System.out.println("Quitting the game...");
                System.exit(1);
            }

            else if(input.equals("I") || input.equals("O") || input.equals("T") || input.equals("J") ||input.equals("L")
             || input.equals("S") || input.equals("Z") || input.equals("R")){
                if(input.equals("R")){
                    Random rand = new Random(); // Creating a random reference.
                    int int_rand = rand.nextInt(7);
                    myShape = shape_arr[int_rand];
                }
                else{
                    myShape = Tetromino.Shapes.valueOf(input);
                }
                Tetromino tetro = new Tetromino(myShape);  //Creating tetromino object using shape input.
                tetBoard.add(tetro);
                tetBoard.draw();

                System.out.println("Tetromino has been moved and rotated randomly...");
                System.out.println("Started lowing the tetromino...");

                tetBoard.animate();
            }     
            else{
                flag= true;
                System.out.println("Invalid  value. Try Again...");
            }

        }
        System.out.println("Game is Over!");
        sc.close();
    
    } // End of main method
} // End of Class

File: 02_Tetromino_Game/driver2.cpp
#include <iostream>
#include "tetris.h"

using namespace std;

int main(){

    int rowSize,columnSize;
    vector<Shapes> shape_arr= {Shapes::I,Shapes::O,Shapes::T,Shapes::J,Shapes::L,Shapes::S,Shapes::Z}; //random array.
    srand(time(NULL));

    cout<<"Enter the row size of the tetris board: ";
    cin>> rowSize;
    cout<<"Enter the column size of the tetris board: ";
    cin>> columnSize;
    Tetris tetris(rowSize,columnSize);  //Creating tetris object using the inputs.

    char input;
    while(1){
        cout<<"Enter the tetromino type(I, O, T, J, L, S, Z) (enter R for random and Q for quit):  ";
        cin>> input;

    Shapes shape = static_cast<Shapes> (input);  //Casting input into shape object.

    if(input== 'Q') {cout<< "Quitting the game.... \n"; exit(1);};
    if(input=='R'){
        shape = static_cast<Shapes> (shape_arr[rand()%7]);
    }
    
    tetris.setTetroType(shape);
    Tetromino tetro(shape);  //Creating tetromino object using shape input.
      
    tetris.initMap();
    tetris.addTetro(tetro);  
    }  

    return 0;
}